api/src/wfl/service/datarepo.clj

```diff
@@ -2,84 +2,99 @@
+      :id))
 
-(defn tabular-ingest
-  "Ingest TABLE at PATH to DATASET-ID in ENVIRONMENT and return the job ID."
-  [environment dataset-id path table]
-  (-> {:format                "json"
-       :ignore_unknown_values true
-       :load_tag              "string"
-       :max_bad_records       0
-       :path                  path
-       :table                 table}
-      (json/write-str :escape-slash false)
-      (->> (thing-ingest environment dataset-id "ingest"))))
+(defn ingest-file
+  "Ingest `source` file as `target` using `dataset-id` and `profile-id`."
+  ([dataset-id profile-id source target options]
+   (-> options
+       (util/assoc-when util/absent? :description (util/basename source))
+       (util/assoc-when util/absent? :mime_type "text/plain")
+       (merge {:profileId profile-id :source_path source :target_path target})
+       (->> (ingest "files" dataset-id))))
+  ([dataset-id profile-id source target]
+   (ingest-file dataset-id profile-id source target {})))
 
-(defn get-job-result
-  "Get result for JOB-ID in ENVIRONMENT."
-  [environment job-id]
-  (let [{:keys [body status]}
-        (http/request
-         {:method :get                 ; :debug true :debug-body true
-          :url (format "%s/jobs/%s/result" (api environment) job-id)
-          :headers (once/get-service-account-header)
-          :throw-exceptions false})]
-    (if (== 200 status)
-      (json/read-str body :key-fn keyword)
-      (throw (HttpException. body)))))
+(defn ingest-dataset
+  "Ingest TABLE at PATH to DATASET-ID and return the job ID."
+  [dataset-id path table]
+  (ingest
+   "ingest"
+   dataset-id
+   {:format                "json"
+    :ignore_unknown_values true
+    :load_tag              "string"
+    :max_bad_records       0
+    :path                  path
+    :table                 table}))
 
 (defn poll-job
   "Return result for JOB-ID in ENVIRONMENT when it stops running."
-  [environment job-id]
-  (letfn [(running? []
-            (-> {:method :get ; :debug true :debug-body true
-                 :url (format "%s/jobs/%s" (api environment) job-id)
-                 :headers (once/get-service-account-header)}
-                http/request :body
-                (json/read-str :key-fn keyword)
-                :job_status
-                #{"running"}))]
-    (when (running?)
-      (Thread/sleep 10000)
-      (poll-job environment job-id))
-    (get-job-result environment job-id)))
+  [job-id]
+  (let [get-result #(-> (repository "jobs/" job-id "/result")
+                        (http/get {:headers (once/get-service-account-header)})
+                        util/response-body-json)
+        running?   #(-> (repository "jobs/" job-id)
+                        (http/get {:headers (once/get-service-account-header)})
+                        util/response-body-json
+                        :job_status
+                        #{"running"})]
+    (while (running?) (.sleep TimeUnit/SECONDS 1))
```

Comment:
Do we want to parameterize the poll interval?

Reply:
Not sure. We can if it causes issues in the future!